climada/hazard/base.py

```diff
@@ -1457,6 +1457,18 @@ def select(self, event_names=None, event_id=None, date=None, orig=None,
                                    dt.datetime.fromordinal(haz.date.min()).year) + 1
             haz.frequency = haz.frequency * year_span_old / year_span_new
 
+        # Check if new fraction is zero everywhere
+        if self.fraction.nnz != 0 and haz.fraction.nnz == 0:
```

Comment:
Another more sensible possible alternative that is potentially easier to read:

```suggestion
        if self.get_fraction() is not None and haz.fraction is None:
```

Reply:
Agreed! One last issue with this: `_get_fraction()` is marked private for some reason, and this will cause an additional linter warning.

Reply:
After personal discussion: Go with `_get_fraction` although the linter will complain 😵